app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/viewmodel/DrawerViewModel.java

```diff
@@ -0,0 +1,128 @@
+package com.teamagam.gimelgimel.app.mainActivity.viewmodel;
+
+import android.support.annotation.NonNull;
+import android.support.design.widget.NavigationView;
+import android.support.v4.widget.DrawerLayout;
+import android.view.MenuItem;
+import android.view.View;
+
+import com.teamagam.gimelgimel.R;
+import com.teamagam.gimelgimel.app.common.base.ViewModels.BaseViewModel;
+import com.teamagam.gimelgimel.app.common.utils.StringToIntegerMapper;
+import com.teamagam.gimelgimel.app.injectors.scopes.PerActivity;
+import com.teamagam.gimelgimel.app.mainActivity.view.MainActivityDrawer;
+import com.teamagam.gimelgimel.domain.layers.entitiy.VectorLayerPresentation;
+import com.teamagam.gimelgimel.domain.map.DisplayVectorLayersInteractor;
+import com.teamagam.gimelgimel.domain.map.DisplayVectorLayersInteractorFactory;
+import com.teamagam.gimelgimel.domain.map.SetVectorLayerVisibilityInteractorFactory;
+
+import javax.inject.Inject;
+
+@PerActivity
+public class DrawerViewModel extends BaseViewModel
+        implements DisplayVectorLayersInteractor.Displayer {
+
+    @Inject
+    DisplayVectorLayersInteractorFactory mDisplayVectorLayersInteractorFactory;
+    @Inject
+    SetVectorLayerVisibilityInteractorFactory mSetVectorLayerVisibilityInteractorFactory;
+
+    private DrawerStateListener mDrawerStateListener;
+
+    private DisplayVectorLayersInteractor mDisplayVectorLayersInteractor;
+
+    private DrawerLayout mDrawerLayout;
+
+    private StringToIntegerMapper mStringToIntegerMapper;
+
+    @Inject
+    public DrawerViewModel() {
+    }
+
+    public void start() {
+        mStringToIntegerMapper = new StringToIntegerMapper();
+        mDrawerStateListener = new DrawerStateListener();
+
+        mDisplayVectorLayersInteractor = mDisplayVectorLayersInteractorFactory.create(this);
+    }
+
+    public void setDrawerMenuSelectedListenerTo(NavigationView navigationView) {
```

Comment:
This method and the next one (setDrawerLayout) are expected to be called.
The class cannot function without them.
Therefore, ask for them at the object's creation.
As usual, either use a static factory that will make all the needed calls, or use AutoFactory for assisted injection